packages/cli/src/commands/build.ts

```diff
@@ -467,6 +468,25 @@ async function doBuild(
       );
       const buildResult = await builder.build(buildOptions);
 
+      if (
+        buildResult &&
+        'output' in buildResult &&
+        'runtime' in buildResult.output &&
+        'type' in buildResult.output &&
+        buildResult.output.type === 'Lambda'
```

Comment:
`buildResult` should always be ok since the code currently doesn't seem to check if it's not ok. If `buildResult` could possibly be not ok, then it should be asserted.

Additionally, optional chaining may make the code more readable:

```
if (buildResult.runtime && buildResult.output?.type === 'Lambda') {
```

Reply:
TS isnt smart enough to use your solution so it needs some help with the type guards. I think if we bump to latest TS it might be slightly better